### ISSUE TITLE
Only use sudo for certain parts of uninstall

### ISSUE THREAD

**Issue by amklose** [kalabox/kalabox#1361]
Bug Report
==========
What is your Kalabox version and operating system?

*v0.12.0-beta1 on Mac OS X El Capitan v 10.11.4*

Have you previously installed Kalabox? If so, what version?

*Yes, I previously had beta1 installed but encountered an error and tried to reinstall*

**Tell us about your bug**

*I did a clean uninstall of kalabox beta1 using the kalabox-uninstall.sh script. Then when I tried installing it again, I encountered the following error:
<img width="625" alt="screenshot 2016-05-20 14 45 44" src="https://cloud.githubusercontent.com/assets/756265/15440566/b9bf3ffa-1e9b-11e6-8998-d28edcd18c4f.png">
*

Does this bug prevent you from using Kalabox?

*Yes*

Here are the releated entries from my install log:
...
May 20 14:50:30 TFTLT42 installd[583]: PackageKit: Executing script "./postinstall" in /private/tmp/PKInstallSandbox.SKuqwy/Scripts/io.kalabox.pkg.engine.yAcEGK
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: Host does not exist: "Kalabox2"
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: Regenerate TLS machine certs?  Warning: this is irreversible. (y/n): Regenerating TLS certificates
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: Host does not exist: "Kalabox2"
May 20 14:50:31 TFTLT42 installd[583]: PackageKit: Executing script "./postinstall" in /private/tmp/PKInstallSandbox.SKuqwy/Scripts/io.kalabox.pkg.services.oG6cUe
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: Host does not exist: "Kalabox2"
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: TLS configuration is invalid - make sure your DOCKER_TLS_VERIFY and DOCKER_CERT_PATH are set correctly.
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: You might need to run `eval "$(docker-machine env default)"`
May 20 14:50:31 TFTLT42 installd[583]: ./postinstall: Something went wrong installing core Docker images
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: releasing backupd
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: allow user idle system sleep
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: Install Failed: Error Domain=PKInstallErrorDomain Code=112 "An error occurred while running scripts from the package “KalaboxInstaller.pkg”." UserInfo={NSFilePath=./postinstall, NSURL=file://localhost/Volumes/Kalabox%201/KalaboxInstaller.pkg#services.pkg, PKInstallPackageIdentifier=io.kalabox.pkg.services, NSLocalizedDescription=An error occurred while running scripts from the package “KalaboxInstaller.pkg”.} {
	    NSFilePath = "./postinstall";
	    NSLocalizedDescription = "An error occurred while running scripts from the package \U201cKalaboxInstaller.pkg\U201d.";
	    NSURL = "file://localhost/Volumes/Kalabox%201/KalaboxInstaller.pkg#services.pkg";
	    PKInstallPackageIdentifier = "io.kalabox.pkg.services";
	}
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: Running idle tasks
May 20 14:50:32 TFTLT42 Installer[86965]: install:didFailWithError:Error Domain=PKInstallErrorDomain Code=112 "An error occurred while running scripts from the package “KalaboxInstaller.pkg”." UserInfo={NSFilePath=./postinstall, NSURL=file://localhost/Volumes/Kalabox%201/KalaboxInstaller.pkg#services.pkg, PKInstallPackageIdentifier=io.kalabox.pkg.services, NSLocalizedDescription=An error occurred while running scripts from the package “KalaboxInstaller.pkg”.}
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: Removing client PKInstallDaemonClient pid=86965, uid=1206961956 (/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer)
May 20 14:50:32 TFTLT42 installd[583]: PackageKit: Done with sandbox removals
May 20 14:50:33 TFTLT42 Installer[86965]: Install failed: The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.
May 20 14:50:33 TFTLT42 Installer[86965]: IFDInstallController EBE29450 state = 8
May 20 14:50:33 TFTLT42 Installer[86965]: Displaying 'Install Failed' UI.
May 20 14:50:33 TFTLT42 Installer[86965]: 'Install Failed' UI displayed message:'The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance.'.

**Comment by amklose** [comment]
Good news - I was able to get the installation to succeed but I found something strange.

I ran the uninstall script again and then opened up VirtualBox and found that the Kalabox2 VM was still running. I powered it off and then went to remove the machine, but the .vdmk file had already been deleted.

After I made sure there were no VMs running and all kalabox files had been removed, I rebooted my computer and tried again with success.

Did I do something wrong initially?

**Comment by pirog** [comment]
Hmmm no that seems correct to me. could possibly be related to https://github.com/kalabox/kalabox/issues/1244

**Comment by danny-englander** [comment]
I just ran into this exact issue on a brand new machine. No luck, even after running the uninstall script. Should I file a brand new issue?

**Comment by pirog** [comment]
@danny-englander can you confirm that the script is not working? after you run it does the kalabox2 vm persist in virtualbox?

**Comment by danny-englander** [comment]
@pirog I guess my case is the reverse of this. I installed Kalabox on a brand new machine so there was nothing to uninstall to begin with, i.e. a previous version of Kalabox. 

So, first, I ran the installer and it failed with the message as in the OP's screen capture above. Then I ran the uninstaller as suggested here but still no luck trying to install again after that.

**Comment by pirog** [comment]
@danny-englander thats definitely interesting. I don't think we've actually seen a failed install on a fresh machine. Usually the install fails on OSX when you've accumulated some legacy and duplicate host only adapters on virtualbox.

**Comment by jserrao** [comment]
@pirog - what would be helpful with the software being in beta would be a documentation page that outlines exactly what Kalabox is installing so we can tweak / check it out. I was looking for this trying to debug on my end, so maybe I just didnt know where to look.

**Comment by pirog** [comment]
@jserrao we have *some* documentation over at https://kalabox.freshdesk.com/solution/categories

That said we know our documentation is pretty weak right now. It's something we are working on.

**Comment by YPCrumble** [comment]
@danny-englander did you figure this out? I'm having the same trouble :/.

**Comment by schutzsmith** [comment]
Same trouble

**Comment by danny-englander** [comment]
@YPCrumble No, I gave up.

**Comment by pirog** [comment]
Our forthcoming release will have a better uninstall script for mac and also some good documentation around uninstalling for those that have trouble.

**Comment by pirog** [comment]
i *think* this should be fixed in the latest `0.13.x` release. If not we've just launched our new docs site over at http://docs.kalabox.io which contains a decent section on manual uninstall for people having issues with the OSX uninstaller.

**Comment by danny-englander** [comment]
This error still persists with beta 3. I ran the new uninstall.sh script as well, still no luck.

**Comment by pirog** [comment]
There should be manual uninstall steps you can follow in the docs.

What shell do you use? Wondering if the script only works with bash.

**Comment by danny-englander** [comment]
I use ZSH / oh-my-zsh in iterm2. I'll try the manual steps. Thanks!

**Comment by pirog** [comment]
Good to know i'll switch over to that shell and see if it fails for me when
i use that.

**Comment by danny-englander** [comment]
Here's the output from the uninstall:

```
Completely remove Kalabox? (Y/N): Y
Removing Kalabox VM...
denglander is not in the sudoers file.  This incident will be reported.
About to remove Kalabox2
Error removing host "Kalabox2": Host does not exist: "Kalabox2"
Can't remove "Kalabox2"
Removing Application...
Removing DNS
All Done!
```

**Comment by pirog** [comment]
oh interesting, so you are not an admin on your machine?

**Comment by danny-englander** [comment]
I'm an admin but not the main admin user. It's a company machine.

**Comment by pirog** [comment]
hmmm strange, i thought all admins had `sudo` access

**Comment by danny-englander** [comment]
I've never had an issue using sudo for other things and I was pretty sure I added myself to the sudoers file but I'll double check.

**Comment by pirog** [comment]
@danny-englander yeah def double check if you can. i'll also definitely explore alternate ways to refactor our uninstall to be more friendly.

**Comment by YPCrumble** [comment]
@danny-englander did you restart your machine? I've been able to do this
successfully but need to restart each time after `sudo
kalabox-uninstall.sh` completes.

**Comment by danny-englander** [comment]
@YPCrumble - I did not try restarting, I will try that. Thanks!

**Comment by danny-englander** [comment]
Restarting did not help matters. [as noted above](https://github.com/kalabox/kalabox/issues/1361#issuecomment-221710427), I am on a brand new machine so there was not ever a previous version of Kalabox installed. Just in case, I followed the manual uninstall steps but there was nothing to do as there was nothing listed in VirtualBox. I also logged into my machine as the root admin and was still not able to install it.
